components/script/dom/xmlhttprequest.rs

```diff
@@ -1051,11 +1051,11 @@ impl XMLHttpRequest {
                 self.upload_complete.set(true);
                 // Substeps 2-4
                 if !self.sync.get() {
-                    self.dispatch_upload_progress_event(atom!("progress"), None);
+                    self.dispatch_upload_progress_event(atom!("progress"), Err(()));
                     return_if_fetch_was_terminated!();
-                    self.dispatch_upload_progress_event(atom!("load"), None);
+                    self.dispatch_upload_progress_event(atom!("load"), Err(()));
```

Comment:
These two lines should use Ok(Some(0)), since this is not an error code path.

Reply:
@jdm thanks for the review. The servo build most of time gets killed because of lack of memory, I have a 8 GB system. I have read somewhere that using llvm linker would use less memory, any idea how to set that up? I have a fedora 30 machine.

Reply:
Based on https://github.com/rust-lang/rust/issues/39915#issuecomment-547866964 and https://github.com/rust-lang/rust/issues/39915#issuecomment-496565005, I think you would need to [follow these instructions](https://github.com/servo/servo/blob/87432c71d0f95f10dbae44bdb080fe51333dd3d5/servobuild.example#L1) and modify the `rustflags` entry to include `-C linker=rust-lld`.